jcontroller.php

```diff
@@ -173,6 +178,34 @@ protected function refRequest($param)
         return $this->refRequest[$param];
     }
 
+    /**
+     * Filters access to an action based on the controller methods.
+     *  
+     * If access is granted, it proceeds to display the action.
+     * If access is not granted, it displays an error code (401 - Access Forbidden).
+     * 
+     * @return void
+     */
+    public function filterAccessControl()
+    {
+        if (!isset($_GET['action'])) {
+            return;
+        }
+
+        $actionName = $_GET['action'];
+        $accessRules = $this->accessRules();
+
+        if (!array_key_exists($actionName, $accessRules)) {
+            return;
+        }
+
+        $accessMethod = $accessRules[$actionName];
+
+        if ($accessMethod != '' && !$this->$accessMethod) {
```

Comment:
This requires us to add separate access methods for each of the actions. But many of the times we use the same access method for all the entire controller other than few exceptions. Please add an option of adding single access method for all actions, for example,

```
[
    'userManagement' => ['User::hasAccess', ['suer-admin']],
    '*' => ['User::hasAccess', ['admin']
] 
```

Meaning this controller can be accessed only by admin users, except userManagement action can only be accessed by `super-admin' users